src/core/js/router.js

```diff
@@ -193,6 +193,17 @@ class Router extends Backbone.Router {
       Adapt.log.deprecated(`Using event based menu view instantiation for '${Adapt.getViewName(model)}'`);
       return;
     }
+
+    if (!isMenu) {
+      // checkIfResetOnRevisit on descendant models before render
+      model.getAllDescendantModels().forEach(model => {
+        if (!model.checkIfResetOnRevisit) return;
+        model.checkIfResetOnRevisit();
+      });
```

Comment:
Just a suggestion but you could shorten this to:
```suggestion
      _.invoke(model.getAllDescendantModels(), 'checkIfResetOnRevisit');
```
(I think `_.invoke` handles checking that the method specified in the 2nd argument actually exists)

Reply:
Well, sheeet. Ain't that pretty.